### ISSUE TITLE
PostgresNode.safe_psql(filename) does not detect syntax errors

### ISSUE THREAD

**Issue by cbandy** [postgrespro/testgres#52]
It took me some time to discover that a syntax error was being not being reported by `node.safe_psql(filename='some.sql')`. I tracked it down to behavior in `psql`. It seems worth raising, though I'm not certain it is something that testgres is necessarily responsible for handling.

---

```
$ psql --version
psql (PostgreSQL) 9.6.4
```
and/or
```
$ psql --version
psql (PostgreSQL) 10.4
```

`psql` reports errors differently depending on how input is passed:

1. ```
   $ psql -h 127.0.0.1 -U cbandy -X -A -t -q -c 'whoops' -p 34751 postgres ; echo $?
   ERROR:  syntax error at or near "whoops"
   LINE 1: whoops
           ^
   1
   ```
2. N.B. exit code is zero here:
   ```
   $ psql -h 127.0.0.1 -U cbandy -X -A -t -q -f <( echo 'whoops' ) -p 34751 postgres ; echo $?
   psql:/dev/fd/63:1: ERROR:  syntax error at or near "whoops"
   LINE 1: whoops
           ^
   0
   ```
   
3. ```
   $ psql -h 127.0.0.1 -U cbandy -X -A -t -q -v ON_ERROR_STOP=1 -f <( echo 'whoops' ) -p 34751 
   postgres ; echo $?
   psql:/dev/fd/63:1: ERROR:  syntax error at or near "whoops"
   LINE 1: whoops
           ^
   3
   ```

**Comment by funbringer** [comment]
Hi Chris,

Thanks for the report! I'll definitely fix this.

**Comment by funbringer** [comment]
This is really odd, but we should provide a workaround nevertheless. I guess it's worth starting a thread in pgsql-hackers.

**Comment by funbringer** [comment]
I found a workaround: if a script starts with `\set ON_ERROR_STOP`, psql exits with non-zero code as soon as it tries to execute a malformed command. Now the question is whether we should add this option to psql's args by default. I'd like to, but this might break some other user's expectations.

**Comment by stalkerg** [comment]
> add this option to psql's args by default

at least it can be optional

> This is really odd

hmm, it's ok if we remember about the client-server architect, in that case for psql it's normal work and this error it's postgres error not psql.

**Comment by ildus** [comment]
I suggest to change environment in `safe_psql`  and keep old behavior in `psql`.

**Comment by funbringer** [comment]
Fixed in 1.8.2.

**Comment by cbandy** [comment]
Thank you for such a quick response!
